src/RateLimiter.php

```diff
@@ -71,38 +72,30 @@ public function __construct(
     /**
      * @inheritdoc
      */
-    public function get($data, ThrottleSettingsInterface $settings = null)
+    public function get($data): ThrottlerInterface
     {
         if (empty($data)) {
-            throw new \InvalidArgumentException('Invalid data, please check the data.');
+            throw new InvalidDataTypeException('Invalid data, please check the data.');
         }
 
         $object = $this->hydratorFactory->make($data)->hydrate($data);
 
         if (!isset($this->throttlers[$object->getKey()])) {
-            $this->throttlers[$object->getKey()] = $this->createThrottler($object, $settings);
+            $this->throttlers[$object->getKey()] = $this->createThrottler($object, $this->defaultSettings);
```

Comment:
This way the settings can no longer be overriden. So for each set of throttlerSettings, a different Ratelimiter instance would be required. I'm missing the advantage of this, but would like to hear your opinion.

Reply:
Yes, this is a things which i think i need to reconsider. The main reason for removing this is because of the conflict it can cause with the previous cache values, when you swap the configuration for the same data (But i think this can be solved by using namespacing the cache values), I'll add this back